### DISCUSSION TITLE
Apprise with Telegram

### DISCUSSION THREAD

**Discussion by ghost** [mcguirepr89/BirdNET-Pi#381] [Q&A] [answered]
Hi there,

I am trying to setup Apprise notifications with telegram. I actually have telegram remote access already up and running for other notification purposes of my Pi's, so i know my bot_token and chat_ids are working.

What i do is go into setting, enter the appropriate tgram-URL (`tgram://*********:***********************************/-*********`) in "Apprise Notifications Configuration:" text field, then hit save. It says: "success", but the text field for "Apprise Notifications Configuration:" goes blank again. I'm not seeing the URL in the field after the page saves/reloads.

As it does not send any notifcations afterwards, i'm feeling like this behaviour is not expected. I would have thought i'm seeing the currently used services, like the URL i just entered, being shown in the field.

I could also not find my tgram-URL anywhere in the birdnet-conf file (but maybe that's intended).
Any hint, what i'm doing wrong here or where i can look to fix it?

Thanks!
Stefan

**Comment by ehpersonal38** [comment, marked as answer]
Hi Stefan,

Sorry to hear you're having trouble with Apprise notifications with Telegram. Could you paste the output of this command so I can further diagnose?

`ls -lh $HOME/BirdNET-Pi/ | grep apprise
`

**Reply by ghost** [reply]
Thank you for asking that question - it lead to the solution:
- found apprise.txt, but it was empty
- upon opening it, i got the sys message "cannot save file, there's no space left"

Apparently all changes to the config i tried to make through the web-frontend didn't go through, as simply the root fs was at 100% full. That's why the text field then stayed blank.

Not only that, it seemed to have stopped working a week ago. I found that $HOME/BirdSongs/Juni-2022 had two directories, 16-Donnerstag/ and 17-Freitag, filling up my disk at both summing up to 23G. Aren't these the temporary working directories?

**Reply by ehpersonal38** [reply]
This could be related to #392 @mcguirepr89

**Reply by ghost** [reply]
Thanks - will follow-up on no-space issue there.

However, i cleaned up both directories, so i could get back to my original apprise problem. Here's the output of your command:

```
birdy117:~/BirdNET-Pi$ ls -lh $HOME/BirdNET-Pi/ | grep apprise
-rw-r--r-- 1 pi pi    0 15. Mai 23:15 apprise.txt
```

The file apprise.txt is empty. When i now, after freeing up space, try to enter the tgram:// URL again, nothing changes. The text-input blanks again and the file stays empty (just mentioning, in case it shouldn't).

**Reply by ehpersonal38** [reply]
Yeah that shouldn't happen - sorry about that. For a quick fix you can `sudo chown caddy:caddy $HOME/BirdNET-Pi/apprise.txt`

I'll look into preventing this in the future

EDIT: I just pushed a commit that should hopefully fix this for the future. 6cee3523b64c2a7adec03c1d3f61f380b743c870

**Reply by ghost** [reply]
That fixed the issue. Apprise notifications to telegram confirmed working. Thanks!

**Comment by srchild** [comment]
Telegram Apprise notifications are working for me. I notice that you precede your chat_id with an underscore before the asterisks. I do not have an underscore there, I just have the 9-digit numeric chat_id

**Reply by ghost** [reply]
Channel IDs may include a hyphen-minus (dash). I have both cases for channel ids (serving multiple channels with a bot) - with and without. The channel id including the hyphen-minus is the correct one.
